test/redis_client_mocks/nearby_cities_search_test.go

```diff
@@ -58,6 +59,10 @@ func TestNearbyCitiesSearch_shouldReturnNearbyCities(t *testing.T) {
 		t.Fatalf("expected number of city returned equals %d, got %d", len(expectedCities), len(resultCities))
 	}
 
+	sort.Slice(resultCities, func(i, j int) bool {
```

Comment:
Does this unit test cover erroneous city retrieval scenario?

Reply:
The tests do not cover the failure scenario. Currently with mini-redis we are doing integration test, so with the interface it is hard to test when we fail to fetch from the database or deserialization.

Reply:
Got it, this is fine then. Do you know if redis provide some mock interface?

Reply:
It does not, that's the point they are making no mocks.